File: library/src/main/java/com/rosberry/abstractrecycler/AbstractViewHolder.java
/*
 *
 *  * Copyright (c) 2018 Rosberry. All rights reserved.
 *
 */

package com.rosberry.abstractrecycler;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @author Alexei Korshun on 12/02/2019.
 */
public class AbstractViewHolder<ITEM extends AbstractItem> extends RecyclerView.ViewHolder {

    public AbstractViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    protected void bind(@NonNull ITEM item) {
    }
}

File: library/src/main/java/com/rosberry/abstractrecycler/AbstractViewHolder.kt
package com.rosberry.abstractrecycler

import android.support.v7.widget.RecyclerView
import android.view.View

/**
 * @author Alexei Korshun on 20/04/2018.
 */
abstract class AbstractViewHolder<in ITEM : AbstractItem>(view: View) : RecyclerView.ViewHolder(view) {

    open fun bind(item: ITEM) {}
}